File: fixtures/scale/ruby/large/lib/check_caesar.rb
require_relative 'caesar_encrypt'
require_relative 'caesar_decrypt'
require_relative 'corpus_additive'

def check_caesar
  corpus_additive.each do |text|
    sealed = caesar_encrypt(text, 3)
    opened = caesar_decrypt(sealed, 3)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_caesar.py
from cipherlab.ciphers.caesar import caesar_encrypt, caesar_decrypt
from cipherlab.specs.caesar_spec import caesar_spec
from cipherlab.corpus.corpus_additive import corpus_additive


def check_caesar():
    spec = caesar_spec()
    for text in corpus_additive():
        sealed = caesar_encrypt(text, spec["key"])
        opened = caesar_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "additive"
